coderd/templates_test.go

```diff
@@ -1809,3 +1810,66 @@ func TestTemplateNotifications(t *testing.T) {
 		})
 	})
 }
+
+func TestTemplateFilterHasAITask(t *testing.T) {
+	t.Parallel()
+
+	db, pubsub := dbtestutil.NewDB(t)
+	client := coderdtest.New(t, &coderdtest.Options{
+		Database:                 db,
+		Pubsub:                   pubsub,
+		IncludeProvisionerDaemon: true,
+	})
+	user := coderdtest.CreateFirstUser(t, client)
+
+	jobWithAI := dbgen.ProvisionerJob(t, db, pubsub, database.ProvisionerJob{
+		OrganizationID: user.OrganizationID,
+		InitiatorID:    user.UserID,
+		Tags:           database.StringMap{},
+		Type:           database.ProvisionerJobTypeTemplateVersionImport,
+	})
+	jobWithoutAI := dbgen.ProvisionerJob(t, db, pubsub, database.ProvisionerJob{
+		OrganizationID: user.OrganizationID,
+		InitiatorID:    user.UserID,
+		Tags:           database.StringMap{},
+		Type:           database.ProvisionerJobTypeTemplateVersionImport,
+	})
+	versionWithAI := dbgen.TemplateVersion(t, db, database.TemplateVersion{
+		OrganizationID: user.OrganizationID,
+		CreatedBy:      user.UserID,
+		HasAITask:      true,
+		JobID:          jobWithAI.ID,
+	})
+	versionWithoutAI := dbgen.TemplateVersion(t, db, database.TemplateVersion{
+		OrganizationID: user.OrganizationID,
+		CreatedBy:      user.UserID,
+		HasAITask:      false,
+		JobID:          jobWithoutAI.ID,
+	})
+	templateWithAI := coderdtest.CreateTemplate(t, client, user.OrganizationID, versionWithAI.ID)
+	templateWithoutAI := coderdtest.CreateTemplate(t, client, user.OrganizationID, versionWithoutAI.ID)
```

Comment:
Let's keep the terminology consistent. This would apply to `versionWith{,out}AI` as well.

```suggestion
	templateWithAITask := coderdtest.CreateTemplate(t, client, user.OrganizationID, versionWithAI.ID)
	templateWithoutAITask := coderdtest.CreateTemplate(t, client, user.OrganizationID, versionWithoutAI.ID)
```